File: shared/src/commonMain/kotlin/com/qualorock/shared/detail/LocationSectionVariant.kt
package com.qualorock.shared.detail

import com.qualorock.shared.domain.Venue

/** DETAIL-004's three-way rule as a pure, testable decision — static map, address-only, or omitted entirely. */
sealed interface LocationSectionVariant {
    data class Map(val url: String, val address: String?) : LocationSectionVariant

    data class AddressOnly(val address: String) : LocationSectionVariant

    data object Omitted : LocationSectionVariant

    companion object {
        fun from(venue: Venue): LocationSectionVariant =
            when {
                venue.staticMapUrl != null -> Map(venue.staticMapUrl, venue.address)
                venue.address != null -> AddressOnly(venue.address)
                else -> Omitted
            }
    }
}

File: iosApp/iosApp/Detail/LocationSectionVariant.swift
import Shared

/// DETAIL-004's three-way rule as a pure, testable decision — static map, address-only, or omitted entirely.
enum LocationSectionVariant: Equatable {
    case map(url: String, address: String?)
    case addressOnly(String)
    case omitted

    static func from(venue: Venue) -> LocationSectionVariant {
        if let mapUrl = venue.staticMapUrl {
            return .map(url: mapUrl, address: venue.address)
        }
        if let address = venue.address {
            return .addressOnly(address)
        }
        return .omitted
    }
}
